lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,16 +16,70 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        binary_path = params[:ipa_path] || params[:apk_path]
+
+        if params[:app] # Set app_id if it is specified as a parameter
+          app_id = params[:app]
+        elsif platform == :ios
+          archive_path = Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]
+          if archive_path
+            app_id = findout_ios_app_id_from_archive(archive_path)
+          end
+        end
+
+        upload_token = get_upload_token(app_id, binary_path)
```

Comment:
You now need to handle the case where `app_id` is null. Maybe print the UI.error! with the `MISSING_APP_ID` message and return?